File: EternalLegacy.UI/src/models/legacy.ts
import { LegacyType } from "./legacyType";
import {DateType} from "./dateType.ts";

export interface ILegacy {
  legacyId: number;
  name: string;
  legacyType: LegacyType;
  openDate?: Date;
  published: boolean;
  legacyName: string;
  dateType: DateType;
}

File: EternalLegacy.API/EternalLegacy.API/ClientContract/Legacy.cs
using EternalLegacy.API.Common;

namespace EternalLegacy.API.ClientContract
{
    public class Legacy
    {
        public int LegacyId { get; set; }
        public string Name { get; set; }
        public LegacyType LegacyType { get; set; }
        public bool IsPublished { get; set; }
        public DateTime? OpenDate { get; set; }
    }
}
